internal/cli/api.go

```diff
@@ -0,0 +1,186 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/auth0/auth0-cli/internal/ansi"
+)
+
+const apiDocsURL = "https://auth0.com/docs/api/management/v2"
+
+var apiFlags = apiCmdFlags{
+	Data: Flag{
+		Name:         "RawData",
+		LongForm:     "data",
+		ShortForm:    "d",
+		Help:         "JSON data payload to send with the request.",
+		IsRequired:   false,
+		AlwaysPrompt: false,
+	},
+}
+
+var apiValidMethods = map[string]bool{
```

Comment:
I ended up changing this to a slice of strings as I do like the fact that we're not repeating ourselves any more for the available methods, however doing a `strings.Join()` all the time is a lot more expensive than the map lookup and as well switching to a for loop to search if the given method is within the valid methods.

>That way, the validation error can be dynamically generated.

The validation error was already dynamically generated even before, could you explain further what do you mean by this?

>Could even take it one step further and use the methods from the http package instead of string literals.

This is a good suggestion 👍🏻 

Updated in: [`2f7766265a` (#531)](https://github.com/auth0/auth0-cli/pull/531/commits/2f7766265a6d993261716ef593b6cd5792d88afe)